sdk/servicebus/service-bus/test/serviceBusClient.spec.ts

```diff
@@ -47,1088 +40,899 @@
+        "Connection string malformed: each part of the connection string must have an `=` assignment.",
+        // "'credentials' is a required parameter and must be an implementation of TokenCredential when using host based constructor overload.",
+        "ErrorMessage is different than expected"
+      );
+    }
+    should.equal(errorWasThrown, true, "Error thrown flag must be true");
+  });
+
+  it("throws error for undefined tokenCredentials", async function(): Promise<void> {
+    try {
+      new ServiceBusClient(serviceBusEndpoint, undefined as any);
+    } catch (err) {
+      errorWasThrown = true;
+      should.equal(
+        err.message,
+        "Connection string malformed: each part of the connection string must have an `=` assignment.",
+        // "'credentials' is a required parameter and must be an implementation of TokenCredential when using host based constructor overload.",
+        "ErrorMessage is different than expected"
+      );
+    }
+    should.equal(errorWasThrown, true, "Error thrown flag must be true");
+  });
+
+  // if (isNode) {
+  //   it("Coerces input to string for host in credential based constructor", async function(): Promise<
+  //     void
+  //   > {
+  //     const tokenCreds = getDefaultTokenCredential();
+  //     sbClient = new ServiceBusClient(123 as any, tokenCreds);
+  //     should.equal(sbClient.name, "sb://123/", "Name of the namespace is different than expected");
+  //   });
+
+  //   it("sends a message to the ServiceBus entity", async function(): Promise<void> {
+  //     const tokenCreds = getDefaultTokenCredential();
+
+  //     const serviceBusClient = createServiceBusClientForTests();
+  //     const entities = await serviceBusClient.test.createTestEntities(
+  //       TestClientType.UnpartitionedQueue
+  //     );
+  //     await serviceBusClient.close();
+
+  //     const sbClient = new ServiceBusClient(serviceBusEndpoint, tokenCreds);
+  //     sbClient.should.be.an.instanceof(ServiceBusClient);
+
+  //     const sender = sbClient.createSender(entities.queue!);
+  //     const receiver = await sbClient.createReceiver(entities.queue!, "peekLock");
+  //     const testMessages = TestMessage.getSample();
+  //     await sender.send(testMessages);
+  //     const msgs = await receiver.receiveBatch(1);
+
+  //     should.equal(Array.isArray(msgs), true, "`ReceivedMessages` is not an array");
+  //     should.equal(msgs[0].body, testMessages.body, "MessageBody is different than expected");
+  //     should.equal(msgs.length, 1, "Unexpected number of messages");
+  //     await sbClient.close();
+  //   });
+  // }
+});
+
+describe("Errors after close()", function(): void {
+  let sbClient: ServiceBusClientForTests;
+  let sender: Sender;
+  let receiver: Receiver<ReceivedMessageWithLock>;
+  let receivedMessage: ReceivedMessageWithLock;
+  let entityName: EntityName;
+  let subscriptionClient: SubscriptionRuleManager;
+
+  afterEach(() => {
+    return sbClient.close();
```

Comment:
I think you want this to be `sbClient.test.afterEach()` in case a test fails so the cleanup still occurs.

Otherwise we'll never be able to get rid of the --exit flag for mocha.

Reply:
Addressed.